crates/db_connection_pool/src/duckdbpool.rs

```diff
@@ -120,10 +89,14 @@ impl DbConnectionPool<r2d2::PooledConnection<DuckdbConnectionManager>, &'static
     }
 }
 
-fn get_duckdb_file(name: &str, params: &Arc<Option<HashMap<String, String>>>) -> String {
-    params
-        .as_ref()
-        .as_ref()
-        .and_then(|params| params.get("duckdb_file").cloned())
-        .unwrap_or(format!("{name}.db"))
+fn get_config(access_mode: &AccessMode) -> Result<duckdb::Config> {
+    let config = duckdb::Config::default()
+        .access_mode(match access_mode {
```

Comment:
Why matching instead of directly passing access_mode into duckdb::Config::default().access_mode() here

Reply:
`access_mode` passed as reference (and in other places above), and can't be dereferenced here